build/index.js

```diff
@@ -35,124 +42,109 @@ const bootstrap = () => {
     if (!fs.existsSync(elmJsonFile)) {
       generateElmJson();
     }
-    return Promise.resolve();
   } catch (e) {
-    return Promise.reject(`Could not complete bootstrap: ${e}`);
+    return new Error(`Could not complete bootstrap: ${e}`);
   }
 };
 
-const fetchIconsArchive = () => {
-  try {
-    const archiveName = "zondicons.zip";
-    const srcArchivePath = `${__dirname}/${archiveName}`;
-    const dstArchivePath = `${tmpPath}/${archiveName}`;
+const wait = (duration) => (...args) => {
+  return new Promise((resolve) => {
+    setTimeout(() => {
+      resolve(...args);
+    }, duration);
+  });
+};
 
-    fs.copyFileSync(srcArchivePath, dstArchivePath);
+const fetchIcons = async () => {
+  const stream = require("stream");
+  const { promisify } = require("util");
+  const fetch = require("node-fetch");
+  const unzipper = require("unzipper");
+  const iconsUrl = "https://www.zondicons.com/zondicons.zip";
+  const asyncStreamPipeline = promisify(stream.pipeline);
 
-    return Promise.resolve(dstArchivePath);
-  } catch (e) {
-    return Promise.reject(`Could not fetch icons archive: ${e}`);
-  }
+  const res = await fetch(iconsUrl);
+
+  return asyncStreamPipeline(res.body, unzipper.Extract({ path: tmpPath }))
```

Comment:
this needs to be `return await ...` here, no?

Reply:
or maybe write it like this ... mixing `async/await` with `Promise.then` is somehow inconsistent
```
await asyncStreamPipeline(res.body, unzipper.Extract({ path: tmpPath }));
await wait(250);
return `${tmpPath}/${basename(iconsUrl)}`;
```

Reply:
:+1:

Reply:
Here it's not wrapped in a try/catch block, so no problem here, but if you do it like that in a try/catch block, the catch block would be ignored

Reply:
True, I try.to avoid try catch pretty strictly and .catch instead.